test/cognitive_job_manager.js

```diff
@@ -126,19 +122,156 @@ contract('CognitiveJobManager', accounts => {
+
+            assertFailureResultCode(result, RESULT_CODE_ADD_TO_QUEUE);
+
+            assert.equal(logEntries, 1, 'Should be fired only 1 event');
+            assert.isOk(logFailure, 'Should be fired failed event');
+            assert.isNotOk(logSuccess, 'Should not be fired successful creation event');
+
+
+            // Erase queue
+            await aliveWorker(workerInstance1, workerOwner1);
+
+            await finishActiveJob(pandora, workerInstance0, workerOwner0);
+
+            assersQueueDepth(pandora, 0);
+
+            assertWorkerState(workerInstance0, WORKER_STATE_ASSIGNED, 0);
+            assertWorkerState(workerInstance1, WORKER_STATE_ASSIGNED, 1);
+
+            await finishActiveJob(pandora, workerInstance0, workerOwner0);
+            await finishActiveJob(pandora, workerInstance1, workerOwner1);
+
+            assertWorkerState(workerInstance0, WORKER_STATE_IDLE, 0);
+            assertWorkerState(workerInstance1, WORKER_STATE_IDLE, 1);
+        });
+
+        it("should create job if it's ok", async () => {
+
+            const active_job_count_start = await pandora.cognitiveJobsCount();
+
+            const result = await createCognitiveJob(pandora, 1);
+
+            const logSuccess = result.logs.filter(l => l.event === 'CognitiveJobCreated')[0];
+            const logFailure = result.logs.filter(l => l.event === 'CognitiveJobCreateFailed')[0];
+            const logEntries = result.logs.length;
+
+            const activeJob = await workerInstance0.activeJob.call();
+
+            const active_job_count_end = await pandora.cognitiveJobsCount();
+
+            assert.equal(
+                active_job_count_end.toNumber(),
+                active_job_count_start.toNumber() + 1,
+                'Active jobs count should be increased');
+
+            assertWorkerState(workerInstance0, WORKER_STATE_ASSIGNED, 0);
+
+            assert.notEqual(activeJob, EMPTY, 'Should set activeJob to worker node');
+
+            assertSuccessResultCode(result, RESULT_CODE_JOB_CREATED);
+
+            assert.equal(logEntries, 2, 'Should be fired only 2 events');
+            assert.isNotOk(logFailure, 'Should not be fired failed event');
+            assert.isOk(logSuccess, 'Should be fired successful creation event');
+
+            await finishActiveJob(pandora, workerInstance0, workerOwner0);
+
+            assertWorkerState(workerInstance0, WORKER_STATE_IDLE, 0);
+            assertWorkerState(workerInstance1, WORKER_STATE_IDLE, 1);
+        });
+
+        it("should not hold payment from customer if job not putted into the queue", async () => {
+            const deposit_before = await pandora.deposits.call(customer);
+
+            await createCognitiveJob(pandora, 1, {from: customer});
+
+            await assersQueueDepth(pandora, 0);
+
+            const deposit_after = await pandora.deposits.call(customer);
+
+            assert.equal(deposit_before.toNumber(), deposit_after.toNumber(), 'Deposit was changed');
```

Comment:
maybe "was NOT changed"? 
Also, you have to check customer's balance (will slightly decrease due gas costs)

Reply:
this case test only deposit changes